modyn/evaluator/internal/metrics/abstract_evaluation_metric.py

```diff
@@ -0,0 +1,46 @@
+from abc import ABC, abstractmethod
+
+import torch
+
+
+class AbstractEvaluationMetric(ABC):
+    """
+    This abstract class is used to represent an evaluation metric which can be used to evaluate a trained model.
+    """
+
+    def __init__(self, progressive: bool):
+        """
+        Abstract class for all evaluation metrics.
+        Currently we only allow progressive metrics i.e metrics where the final evaluation value can be calculated
+        on-the-fly without having the model output from all batches.
+
+        Args:
+            progressive: whether the metrics is progressive.
+        """
+        self.progressive = progressive
```

Comment:
usually, you say a metric is _decomposable_ for what you call progressive. non-decomposable metrics (like median) are also called holistic. we could either flip the meaning and call this holistic or rename progressive to decomposable (I'd probably do that)

Reply:
I see, thank you for clarification! In the next commit I did some changes to reflect this.